src/test/java/net/sf/jabref/model/database/BibDatabaseTest.java

```diff
@@ -187,4 +189,29 @@ public void addSameStringIdTwiceThrowsKeyCollisionException() {
         database.addString(string);
         fail();
     }
+
+    @Test
+    public void testAddEntryEventReceivement() {
+        BibDatabase database = new BibDatabase();
+        BibEntry shouldBeEntry = new BibEntry(IdGenerator.next());
+        TestEventListener tel = new TestEventListener();
+        EventBus eventBus = database.getEventBus();
+        eventBus.register(tel);
+        database.insertEntry(shouldBeEntry);
+        BibEntry isEntry = tel.getBibEntry();
+        assertEquals(shouldBeEntry, isEntry);
+    }
+
+    @Test
+    public void testRemoveEntryEventReceivement() {
+        BibDatabase database = new BibDatabase();
+        BibEntry shouldBeEntry = new BibEntry(IdGenerator.next());
+        TestEventListener tel = new TestEventListener();
+        EventBus eventBus = database.getEventBus();
+        eventBus.register(tel);
+        database.insertEntry(shouldBeEntry);
```

Comment:
insertEntry and removeEntry are two different events. The event listener should be reset inbetween. Otherwise, the local variable storing the bibentry might not be changed.

Adapt (and rename) this combined test and add another test testing the removal only.